src/Widgets/IconChooser.vala

```diff
@@ -0,0 +1,89 @@
+const double M_PI = 3.14159265358979323846;
+
+class IconChooser: Gtk.EventBox {
+    private Gdk.Pixbuf pixbuf;
+
+    public string icon_path { get; set; }
+
+    public IconChooser(string icon_path) {
+        Object ( icon_path: icon_path );
+        this.load_icon(icon_path);
+    }
+
+    construct {
+        Gtk.DrawingArea drawing_area = new Gtk.DrawingArea();
+        drawing_area.halign = Gtk.Align.CENTER;
+        drawing_area.hexpand = true;
+        drawing_area.set_size_request(128, 128);
+        drawing_area.draw.connect((widget, cr) => {
+            int width = this.pixbuf.get_width ();
+            int height = this.pixbuf.get_height ();
+            cr.arc(64.0, 64.0, 64, 0, 2*M_PI);
+            cr.clip();
+            cr.scale(128.0 / (double) width, 128.0 / (double) height);
+            Gdk.cairo_set_source_pixbuf(cr, this.pixbuf, 1, 1);
+            cr.paint();
```

Comment:
It might make sense to use [Hdy.Avatar](https://valadoc.org/libhandy-1/Hdy.Avatar.html) here instead of manually calculating a circle and clipping this.

Reply:
Not quite sure how to use it. At least in this documentation, I'm not seeing a way to load an image by path or pixbuf.

Reply:
It looks like `HdyAvatar` uses `GLoadableIcon`, an interface implemented by `GFileIcon`, which can be created with a `GFile`.

(Perhaps libhandy could use some helper functions to make that simpler in simple use cases.)

Reply:
Now I see where the classes were hidden. Trying it out.

Reply:
Unfortunately, it's failing to set an icon without any errors being propagated.

```vala
this.avatar.set_loadable_icon (new GLib.FileIcon (GLib.File.new_for_path (path)));
```

Visually, it looks and functions identically to what we have, so I don't think we need to import libhandy for this. Drawing a clipped circular image with cairo is simple enough as is.